ai_transform/engine/abstract_engine.py

```diff
@@ -337,6 +334,46 @@ def update_chunk(
 
             raise MaxRetriesError("max number of retries exceeded")
 
+    def api_progress(
+        self,
+        iterator: Iterator,
+        show_progress_bar: bool = True,
+        n_total: int = None,
+        n_passes: int = 1,
+        pass_index: int = 0,
+    ) -> Iterator:
+        assert n_passes >= 1, "`n_passes` must be strictly positive and greater than 0"
```

Comment:
this is an odd coding convention - this should just be placed in the thing above.